pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -218,6 +190,12 @@ sys.exit('no disk found with OSD ID $OSD_ID')
 	# ceph-volume raw mode only supports bluestore so we don't need to pass a store flag
 	ceph-volume raw activate --device "$DEVICE" --no-systemd --no-tmpfs
 fi
+
+# something in the 'ceph-volume activate' command resets the key to a prior version, so refresh the
+# keyring file again at the end of activate init
+ceph -n client.admin auth get-or-create osd."$OSD_ID" \
+  mon 'allow profile osd' mgr 'allow profile osd' osd 'allow *' \
+  -k /etc/ceph/admin-keyring-store/keyring > "$KEYRING_FILE"
```

Comment:
So this is really strange. I looked into `ceph-volume raw activate`, and it doesn't seem to be touching the keyring at all. It does call `ceph-bluestore-tool` which I suspect is grabbing the key from the bluestore contents. In my testing, refreshing the key at the end of 'activate' resolves things, at least for raw OSDs on hosts.

I still have to test PVC OSDs, and I suspect the OSD bootstrap keyring may have issues given its current implementation.